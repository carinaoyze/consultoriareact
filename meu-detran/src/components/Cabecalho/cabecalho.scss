.cabecalho-site {
    background: url('../../assets/images/bg-header.png') no-repeat center center;

    &::before {
        content: '';
        display: block;
        background: #00A551;
        width: 380px;
        height: 5px;
        left: 0;
        top: 0;
        position: absolute;
    }

    .container-fluid {
        background-color: rgba(22, 40, 70, 0.9);

        h2, h3, h4 {
            font-weight: 700;
            text-transform: uppercase;
        }
    
        h2 {
            font-size: 32px;
            letter-spacing: .18rem;
    
            a {
                text-decoration: none;
                color: white;
    
                span {
                    color: #00A551;
                }
            }
        }
    
        h3 {
            font-size: 16px;
            letter-spacing: .34rem;
        }
    
        h4 {
            font-size: 10px;
            color: #9CADC7;
            font-weight: normal;
            letter-spacing: .095rem;
        }
    
        button {
            text-transform: uppercase;
            font-size: 14px;
        }
    }
}