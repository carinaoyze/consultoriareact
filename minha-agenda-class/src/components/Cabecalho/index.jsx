import React, { Component } from 'react';

class Cabecalho extends Component {
    render() {
        return (
            <div className="jumbotron">
                <h1 className="display-4">{ this.props.titulo }</h1>
                <p className="lead">
                    { this.props.texto }
                </p>
            </div>
        );
    }
}

export default Cabecalho;