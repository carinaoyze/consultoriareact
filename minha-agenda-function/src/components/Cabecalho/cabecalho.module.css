.cabecalho {
    background-color: red;
    color: white;
}

.textoEspecial {
    color: green;
}