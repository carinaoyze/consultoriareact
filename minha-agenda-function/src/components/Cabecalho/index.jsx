import React from 'react';
import cabecalhoStyles from './cabecalho.module.css';

function Cabecalho({ titulo, texto }) {
    return (
        <div className={`jumbotron ${cabecalhoStyles.cabecalho}`}>
            <h1 className="display-4">{ titulo }</h1>
            <p className="lead">
                { texto }
            </p>
        </div>
    );
}

export default Cabecalho;