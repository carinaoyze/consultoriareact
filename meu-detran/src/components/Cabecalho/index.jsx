import React from 'react';
import { Container, Row, Col } from 'react-bootstrap';
import { Link } from 'react-router-dom';
import ModalContato from '../Modals/ModalContato';
import ModalCadastro from '../Modals/ModalCadastro';
import ModalLogin from '../Modals/ModalLogin';

import logoMeuDetran from '../../assets/images/detran-ms.png';
import './cabecalho.scss';

export default function Cabecalho() {
    return (
        <header className="cabecalho-site position-relative">
            <Container fluid className="py-2 px-md-4 px-xl-5">
                <Row className="align-items-center pb-2 pt-3 mb-1">
                    <Col sm="auto" className="pr-0">
                        <img src={ logoMeuDetran } alt="Meu Detran" />
                    </Col>
                    <Col className="pr-sm-0 pr-md-3">
                        <h2 className="mb-0">
                            <Link to="/">
                                Meu <span>Detran</span>
                            </Link>
                        </h2>
                        <h3 className="text-white mb-1">Serviços Digitais</h3>
                        <h4 className="mb-2">Departamento Estadual de Trânsito</h4>
                    </Col>
                    <Col sm={6} md={5} lg={3} className="ml-auto mt-3 mt-sm-0">
                        <Row className="mb-2">
                            <Col>
                                <ModalContato />
                            </Col>
                        </Row>
                        <Row>
                            <Col>
                                <ModalCadastro />
                            </Col>
                            <Col className="pl-0">
                                <ModalLogin />
                            </Col>
                        </Row>
                    </Col>
                </Row>
            </Container>
        </header>
    )
}
