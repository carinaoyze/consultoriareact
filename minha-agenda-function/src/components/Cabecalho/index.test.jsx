import { render } from '@testing-library/react';
import Cabecalho from './index';

describe('Componente Cabeçalho', () => {
    test('Quando o cabeçalho for renderizado, deve receber as propriedades corretamente', () => {
        const { container } = render( 
            <Cabecalho 
                titulo="Minha Agenda" 
                texto="Confira abaixo sua lista de contatos cadastrados." 
            /> 
        );
        expect(container.firstChild).toMatchSnapshot();
    });
});