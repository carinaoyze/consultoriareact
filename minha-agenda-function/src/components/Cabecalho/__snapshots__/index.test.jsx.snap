// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Componente Cabeçalho Quando o cabeçalho for renderizado, deve receber as propriedades corretamente 1`] = `
<div
  class="jumbotron cabecalho"
>
  <h1
    class="display-4"
  >
    Minha Agenda
  </h1>
  <p
    class="lead"
  >
    Confira abaixo sua lista de contatos cadastrados.
  </p>
</div>
`;
